libifupdown/lifecycle.c

```diff
@@ -457,8 +460,17 @@ lif_lifecycle_run(const struct lif_execute_opts *opts, struct lif_interface *ifa
 		if (!lif_lifecycle_run_phase(opts, iface, "create", lifname, up))
 			return false;
 
+		if (!lif_waitif_setup(&wif, lifname))
+			return false;
 		if (!lif_lifecycle_run_phase(opts, iface, "pre-up", lifname, up))
 			return false;
+		if (opts->verbose)
+			fprintf(stderr, "ifupdown: waiting for interface %s to change to up state\n", lifname);
+		if (!lif_waitif_wait(&wif, 10)) { /* TODO: Make timeout configurable */
```

Comment:
TODO: Make the timeout configurable, maybe via a `ifup(8)` command-line flag (i.e. analog to `-T`)?